runtime/src/lib.rs

```diff
@@ -1039,13 +1111,55 @@ impl_runtime_apis! {
                     weights: stats.weights,
                 },
                 params: ModuleParams {
-                    name: params.name,
-                    address: params.address,
+                    name: String::from_utf8(params.name).expect("Name bytes should be valid utf8"),
+                    address: String::from_utf8(params.address).expect("Name bytes should be valid utf8"),
                     delegation_fee: params.delegation_fee,
                     controller: params.controller,
                 }
             }
         }
+
+        fn get_key_info(key: AccountId) -> KeyInfo {
+            let balance = SubspaceModule::get_balance_u64(&key);
+            let total_stake = SubspaceModule::get_global_stake_to(&key);
+            let mut stake_to: Vec<KeyStakeToInfo> = Vec::new();
+
+            for netuid in SubspaceModule::netuids() {
+                let  mut stake_to_module: Vec<(String, u64)> = Vec::new();
+
+                let stake_to_vec = SubspaceModule::get_stake_to_vector(netuid, &key);
+
+                if !stake_to_vec.is_empty() {
+                    for (module_key, amount) in stake_to_vec.iter() {
+                        let module_name = SubspaceModule::module_params(netuid, module_key).name;
+
+                        if *amount > 0 {
+                            stake_to_module.push((
+                                String::from_utf8(module_name).expect("Name bytes should be valid utf8"),
+                                *amount,
+                            ))
+                        }
+                    }
+                }
```

Comment:
There's no need to get the module name if the amount is zero. The code can also be a bit more organized, don't you think?
```suggestion
                let stake_to_module = stake_to_vec
                    .into_iter()
                    .filter(|(_, &a)| a > 0)
                    .map(|(mod_key, amount)| {
                        let name = SubspaceModule::get_module_name(netuid, module_key);
                        let name = String::from_utf8(name).expect("Name must be valid utf8");
                        (name, amount)
                    })
                    .collect();
```
I'm not sure `get_module_name` exists, but something similar would be ideal.